server/sonar-server/src/test/java/org/sonar/server/computation/step/ComputeQProfileMeasureStepTest.java

```diff
@@ -76,47 +97,27 @@
   @Rule
   public MeasureRepositoryRule measureRepository = MeasureRepositoryRule.create(treeRootHolder, metricRepository);
 
+  @Rule
+  public AnalysisMetadataHolderRule analysisMetadataHolder = new AnalysisMetadataHolderRule();
+
   ComputeQProfileMeasureStep underTest;
 
   @Before
   public void setUp() throws Exception {
-    underTest = new ComputeQProfileMeasureStep(treeRootHolder, measureRepository, metricRepository);
+    underTest = new ComputeQProfileMeasureStep(treeRootHolder, measureRepository, metricRepository, analysisMetadataHolder);
   }
 
   @Test
   public void add_quality_profile_measure_on_project() throws Exception {
     treeRootHolder.setRoot(MULTI_MODULE_PROJECT);
-
-    QualityProfile qp = createQProfile(QP_NAME_1, LANGUAGE_KEY_1);
-    addMeasure(SUB_MODULE_REF, qp);
+    QualityProfile qpJava = createQProfile(QP_NAME_1, LANGUAGE_KEY_1);
+    QualityProfile qpPhp = createQProfile(QP_NAME_2, LANGUAGE_KEY_2);
+    analysisMetadataHolder.setQProfilesByLanguage(ImmutableMap.of(LANGUAGE_KEY_1, qpJava, LANGUAGE_KEY_2, qpPhp));
 
     underTest.execute();
 
-    assertThat(measureRepository.getAddedRawMeasures(PROJECT_REF).get(QUALITY_PROFILES_KEY)).extracting("data").containsOnly(toJson(qp));
-  }
-
-  @Test
-  public void add_quality_profile_measure_from_multiple_modules() throws Exception {
```

Comment:
Previsoulsy there was a special case for modules (need to merge QP measure) but this is no more the case.